Userland/Libraries/LibJS/Runtime/Temporal/Duration.cpp

```diff
@@ -436,6 +441,490 @@ ThrowCompletionOr<BalancedDuration> balance_duration(GlobalObject& global_object
+    // 5. Let days be ? DaysUntil(relativeTo, later).
+    auto days = days_until(global_object, relative_to, *later);
+
+    // 6. Let dateTime be ? CreateTemporalDateTime(later.[[ISOYear]], later.[[ISOMonth]], later.[[ISODay]], relativeTo.[[ISOHour]], relativeTo.[[ISOMinute]], relativeTo.[[ISOSecond]], relativeTo.[[ISOMillisecond]], relativeTo.[[ISOMicrosecond]], relativeTo.[[ISONanosecond]], relativeTo.[[Calendar]]).
+    auto* date_time = TRY(create_temporal_date_time(global_object, later->iso_year(), later->iso_month(), later->iso_day(), relative_to.iso_hour(), relative_to.iso_minute(), relative_to.iso_second(), relative_to.iso_millisecond(), relative_to.iso_microsecond(), relative_to.iso_nanosecond(), relative_to.calendar()));
+
+    // 7. Return the Record { [[RelativeTo]]: dateTime, [[Days]]: days }.
+    return MoveRelativeDateResult { .relative_to = make_handle(date_time), .days = days };
+}
+
+// 7.5.17 MoveRelativeZonedDateTime ( zonedDateTime, years, months, weeks, days ), https://tc39.es/proposal-temporal/#sec-temporal-moverelativezoneddatetime
+ThrowCompletionOr<ZonedDateTime*> move_relative_zoned_date_time(GlobalObject& global_object, ZonedDateTime& zoned_date_time, double years, double months, double weeks, double days)
+{
+    // 1. Let intermediateNs be ? AddZonedDateTime(zonedDateTime.[[Nanoseconds]], zonedDateTime.[[TimeZone]], zonedDateTime.[[Calendar]], years, months, weeks, days, 0, 0, 0, 0, 0, 0).
+    auto* intermediate_ns = TRY(add_zoned_date_time(global_object, zoned_date_time.nanoseconds(), &zoned_date_time.time_zone(), zoned_date_time.calendar(), years, months, weeks, days, 0, 0, 0, 0, 0, 0));
+
+    // 2. Return ! CreateTemporalZonedDateTime(intermediateNs, zonedDateTime.[[TimeZone]], zonedDateTime.[[Calendar]]).
+    return MUST(create_temporal_zoned_date_time(global_object, *intermediate_ns, zoned_date_time.time_zone(), zoned_date_time.calendar()));
+}
+
+// 7.5.18 RoundDuration ( years, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds, increment, unit, roundingMode [ , relativeTo ] ), https://tc39.es/proposal-temporal/#sec-temporal-roundduration
+ThrowCompletionOr<RoundedDuration> round_duration(GlobalObject& global_object, double years, double months, double weeks, double days, double hours, double minutes, double seconds, double milliseconds, double microseconds, double nanoseconds, u32 increment, StringView unit, StringView rounding_mode, Object* relative_to_object)
+{
+    // NOTE: `relative_to_object`, `relative_to_date`, and `relative_to` in the various code paths below
+    // are all the same as far as the spec is concerned, but the latter two are more strictly typed for convenience.
+    // The `_date` suffix is used as relativeTo is guaranteed to be a PlainDateTime object or undefined after step 5
+    // (i.e. PlainDateTime*), but a PlainDate object is assigned in a couple of cases.
+    PlainDateTime* relative_to = nullptr;
+
+    auto& vm = global_object.vm();
```

Comment:
nitpick: I'd move relative_to (and the comment) to after step 1 (since they reference each other), which will also make `auto& vm` first like other LibJS methods

Reply:
great point, done!